metrics.go

```diff
@@ -39,26 +46,43 @@ func NewMetrics() *Metrics {
 			[]string{"topic"},
 		),
 	}
+
+	if err := m.registry.Register(m.subscribers); err != nil {
```

Comment:
This doesn't work. Even if `unregister()` has been called, it's not possible to register the same metric twice...

Reply:
And because how Caddy works, we need to find a workaround because when the config change (and in many other cases), this method is called again.

Reply:
IMHO `Metrics` should be a dependency of the Hub as well. Then you can create it upfront.